batching.go

```diff
@@ -9,27 +9,45 @@ import (
 )
 
 type batch struct {
-	ds    *Datastore
-	batch *pgx.Batch
+	ds           *Datastore
+	batch        *pgx.Batch
+	maxBatchSize uint16
 }
 
 // Batch creates a set of deferred updates to the database.
 func (d *Datastore) Batch() (ds.Batch, error) {
-	return &batch{ds: d, batch: &pgx.Batch{}}, nil
+	b := &batch{ds: d, batch: &pgx.Batch{}, maxBatchSize: 0}
```

Comment:
Should we set the default `maxBatchSize` to 1 since that emulates the current behavior?